### PR TITLE
rework edit position/department endpoint

### PR DESCRIPTION
**Pull request by BelehovEgor**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/719

### CODE REVIEW

File: src/CompanyService.Data/OfficeRepository.cs [hunk 1]
```diff
@@ -36,9 +36,9 @@ public List<DbOffice> Find(int skipCount, int takeCount, out int totalCount)
                 throw new BadRequestException("Take count can't be equal or less than 0.");
             }
 
-            totalCount = _provider.Offices.Count(x => x.IsActive);
+            totalCount = _provider.Offices.Count();
 
-            return _provider.Offices.Skip(skipCount * takeCount).Take(takeCount).Where(o => o.IsActive).ToList();
+            return _provider.Offices.Skip(skipCount * takeCount).Take(takeCount).ToList();
```
Comment (MariyaSin) [inline src/CompanyService.Data/OfficeRepository.cs:41]: Please, remove takeConunt from skip arguments

File: src/CompanyService.Data/OfficeRepository.cs [hunk 2]
```diff
@@ -36,9 +36,9 @@ public List<DbOffice> Find(int skipCount, int takeCount, out int totalCount)
                 throw new BadRequestException("Take count can't be equal or less than 0.");
             }
 
-            totalCount = _provider.Offices.Count(x => x.IsActive);
+            totalCount = _provider.Offices.Count();
```
Comment (MariyaSin) [inline src/CompanyService.Data/OfficeRepository.cs:39]: I think you should add IsActive check

**Review by MariyaSin — APPROVED** [review]

File: src/CompanyService.Data/OfficeRepository.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/CompanyService.Data/OfficeRepository.cs:41]: fixed

File: src/CompanyService.Data/OfficeRepository.cs (hunk 2 above)
Comment (BelehovEgor) [reply src/CompanyService.Data/OfficeRepository.cs:39]: fixed

File: src/CompanyService.Data/OfficeRepository.cs (hunk 1 above)
Comment (TIove) [reply src/CompanyService.Data/OfficeRepository.cs]: ?

**Review by TIove — APPROVED** [review]

File: src/CompanyService.Data/OfficeRepository.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/CompanyService.Data/OfficeRepository.cs]: new rules

File: src/CompanyService.Data/OfficeRepository.cs [hunk 3]
```diff
@@ -29,16 +29,23 @@ public void Add(DbOffice office)
...
+        public List<DbOffice> Find(int skipCount, int takeCount, bool? includeDeactivated, out int totalCount)
         {
             if (takeCount <= 0)
             {
                 throw new BadRequestException("Take count can't be equal or less than 0.");
             }
 
-            totalCount = _provider.Offices.Count(x => x.IsActive);
+            if (includeDeactivated.HasValue && includeDeactivated.Value)
+            {
```
Comment (djino97) [inline src/CompanyService.Data/OfficeRepository.cs:40]: Where use _provider.Offices.AsQueryable() ?

File: src/CompanyService.Validation/Position/EditPositionRequestValidator.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            #endregion
+
+            #region Description
+
+            AddFailureForPropertyIf(
+                nameof(EditPositionRequest.Description),
+                x => x == OperationType.Replace,
+                new()
+                {
+                    { x => !string.IsNullOrEmpty(x.value.ToString()), "Description is too short." },
```
Comment (djino97) [inline src/CompanyService.Validation/Position/EditPositionRequestValidator.cs:53]: message -- Description should not be empty.

File: src/CompanyService.Validation/Position/EditPositionRequestValidator.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            #endregion
+
+            #region Name
+
+            AddFailureForPropertyIf(
+                nameof(EditPositionRequest.Name),
+                x => x == OperationType.Replace,
+                new()
+                {
+                    { x => !string.IsNullOrEmpty(x.value.ToString()), "Name is too short." },
```
Comment (djino97) [inline src/CompanyService.Validation/Position/EditPositionRequestValidator.cs:41]: message -- Name should not be empty.

**Review by djino97 — APPROVED** [review]

**Review by MariyaSin — APPROVED** [review]

**Review by TIove — APPROVED** [review]

File: src/CompanyService.Validation/Position/EditPositionRequestValidator.cs (hunk 4 above)
Comment (BelehovEgor) [reply src/CompanyService.Validation/Position/EditPositionRequestValidator.cs:53]: fixed

File: src/CompanyService.Validation/Position/EditPositionRequestValidator.cs (hunk 5 above)
Comment (BelehovEgor) [reply src/CompanyService.Validation/Position/EditPositionRequestValidator.cs:41]: fixed

File: src/CompanyService.Data/OfficeRepository.cs (hunk 3 above)
Comment (BelehovEgor) [reply src/CompanyService.Data/OfficeRepository.cs:50]: fixed

**Pull request merged**